src/mlpack/tests/random_forest_test.cpp

```diff
@@ -483,4 +456,32 @@ BOOST_AUTO_TEST_CASE(RandomForestCategoricalTrainReturnEntropy)
   BOOST_REQUIRE_EQUAL(std::isfinite(entropy), true);
 }
 
+/**
+ * Test that different trees get generated.
+  */
+BOOST_AUTO_TEST_CASE(DifferentTreesTest)
+{
+  arma::mat d(10, 100, arma::fill::randu);
+  arma::Row<size_t> l(100);
+  for (size_t i = 0; i < 50; ++i)
+    l(i) = 0;
+  for (size_t i = 50; i < 100; ++i)
+    l(i) = 1;
+
+  bool success = false;
+  size_t trial = 0;
+
+  while (!success && trial < 10)
```

Comment:
I think we should clarify the multiple trials here.

Reply:
Ah, 10 is too many I think, but the trials are still useful.  Comment fixed in cc7e2accb.